tests/functional/flakes/develop.sh

```diff
@@ -0,0 +1,70 @@
+source ../common.sh
+
+clearStore
+rm -rf $TEST_HOME/.cache $TEST_HOME/.config $TEST_HOME/.local
+
+# Create flake under test.
+cp ../shell-hello.nix ../config.nix $TEST_HOME/
+cat <<EOF > $TEST_HOME/flake.nix
+{
+    inputs.nixpkgs.url = "$TEST_HOME/nixpkgs";
+    outputs = {self, nixpkgs}: {
+      packages.$system.hello = (import ./config.nix).mkDerivation {
+        name = "hello";
+        outputs = [ "out" "dev" ];
+        meta.outputsToInstall = [ "out" ];
+        buildCommand = "";
+      };
+    };
+}
+EOF
+
+# Create fake nixpkgs flake.
+mkdir -p $TEST_HOME/nixpkgs
+cp ../config.nix ../shell.nix $TEST_HOME/nixpkgs
+cat <<EOF > $TEST_HOME/nixpkgs/flake.nix
+{
+    outputs = {self}: {
+      legacyPackages.$system.bashInteractive = (import ./shell.nix {}).bashInteractive;
+    };
+}
+EOF
+
+cd $TEST_HOME
+
+# Test whether `nix develop` passes through environment variables.
+[[ "$(
+    ENVVAR=a nix develop --no-write-lock-file .#hello <<EOF
+echo "\$ENVVAR"
+EOF
+)" = "a" ]]
+
+# Test whether `nix develop --ignore-environment` does _not_ pass through environment variables.
+[[ -z "$(
+    ENVVAR=a nix develop --ignore-environment --no-write-lock-file .#hello <<EOF
+echo "\$ENVVAR"
+EOF
+)" ]]
+
+# Determine the bashInteractive executable.
+nix build --no-write-lock-file './nixpkgs#bashInteractive' --out-link ./bash-interactive
+BASH_INTERACTIVE_EXECUTABLE="$PWD/bash-interactive/bin/bash"
+
+# Test whether `nix develop` uses nixpkgs#bashInteractive shell.
+[[ "$(
+    nix develop --ignore-environment --no-write-lock-file .#hello <<EOF
+export PATH="$PATH"
+readlink /proc/\$$/exe
+EOF
+)" -ef "$BASH_INTERACTIVE_EXECUTABLE" ]]
+
+# Test whether `nix develop` sets `SHELL` to nixpkgs#bashInteractive shell.
+[[ "$(
+    SHELL=custom nix develop --ignore-environment --no-write-lock-file .#hello <<EOF
```

Comment:
You might want to duplicate these tests to ensure they pass without `--ignore-environment` as well. I also see the tests are failing on macOS, I'll have a look at that.

Reply:
@bobvanderlinden OK I found it, macOS doesn't have procfs. Try applying this patch:

```diff
diff --git a/tests/functional/flakes/develop.sh b/tests/functional/flakes/develop.sh
index abfa67da9..a59dbca2b 100644
--- a/tests/functional/flakes/develop.sh
+++ b/tests/functional/flakes/develop.sh
@@ -54,7 +54,7 @@ BASH_INTERACTIVE_EXECUTABLE="$PWD/bash-interactive/bin/bash"
 [[ "$(
     nix develop --ignore-environment --no-write-lock-file .#hello <<EOF
 export PATH="$PATH"
-readlink /proc/\$$/exe
+lsof -a -d txt -p $$ 2>&1 | grep -o '/.*/bash'
 EOF
 )" -ef "$BASH_INTERACTIVE_EXECUTABLE" ]]
 ```

Reply:
> You might want to duplicate these tests to ensure they pass without `--ignore-environment` as well. I also see the tests are failing on macOS, I'll have a look at that.

Good one. Yes I'll add it.

> OK I found it, macOS doesn't have procfs.

Doh, of course. Good find and thanks for the suggestion! I'll have to rewrite it to use ps instead of lsof to avoid having to add lsof to the test environment.

Reply:
What version of ps are you using? The one I have, which seems to be what the tests use as well, neither has the `--pid` not the `--format` option. In fact, it can't even parse double dashes:

```
$ ps --pid $$ --format exe --no-headers
/bin/ps: illegal option -- -
usage: ps [-AaCcEefhjlMmrSTvwXx] [-O fmt | -o fmt] [-G gid[,gid...]]
          [-g grp[,grp...]] [-u [uid,uid...]]
          [-p pid[,pid...]] [-t tty[,tty...]] [-U user[,user...]]
       ps [-L]
```

Maybe adding lsof to the test environment isn't so bad?